### PR TITLE
feat(M1): QR generation + check-in activation flow (KIM-316, KIM-324)

### PR DESCRIPTION
**Pull request by Oiranca**
## Summary

- Extends `reservation_status` enum with `pending` and `no_show` values; reservations now start as `pending` and transition to `active` on QR scan
- Generates unique QR codes (absolute URL) on table creation; double (`removable_top`) tables get two QRs — main and INF side
- New `POST /api/tables/[id]/activate` endpoint with 20-minute activation window validation
- New `POST /api/tables/[id]/qr` endpoint for admin QR regeneration (admin only)
- New `/check-in/[tableId]` page with auth guard and per-state result UI
- Admin rooms section now shows QR image + download button per table
- Status badges extended for `pending` and `no_show` in reservations and admin views
- Conflict detection extended to include `pending` reservations (prevents double-booking stacking)
- Non-admin users blocked from setting `status = active` directly (must use QR flow)

## Security fixes applied (from review cycle 1)
- Block `status → active` transition for non-admin via `updateReservationForSession`
- Include `pending` in `listActiveReservationsForConflict`
- QR payload uses absolute URL (`NEXT_PUBLIC_APP_URL`) so native camera apps auto-navigate

## Deferred to hardening issues
- Timezone anchoring for activation window
- TOCTOU race condition in activation (atomic conditional UPDATE)
- UUID validation on `tableId` URL param
- Rate limit on admin QR regeneration endpoint
- Surface filter for non-`removable_top` tables (`side` logic)

## Test plan
- [x] 235 tests pass (`pnpm test`) — tests added for cron routes, check-in activator, QR storage upload, and reservation activation
- [x] `pnpm typecheck` passes
- [x] `pnpm build` passes — no ESLint warnings or errors
- [x] `activateReservationByTable` covered: success, all 4 error codes, `side=inf`, boundary conditions
- [x] QR generation tests: absolute URL format, removable-top dual QR, Storage upload path
- [x] i18n parity verified: `checkin.*`, `reservations.pending`, `reservations.no_show`
- [x] Cron route: GET and POST both covered, auth guard tested

Closes #KIM-316
Closes #KIM-324

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: lib/server/reservations-service.ts [hunk 1]
```diff
@@ -388,3 +394,91 @@ export async function updateReservationForSession(
...
+  single: () => Promise<{ data: ReservationRow | null; error: PostgrestErrorLike | null }>
+}
+
+export async function activateReservationByTable(
+  tableId: string,
+  userId: string,
+  side?: 'inf',
+): Promise<Reservation> {
+  const today = new Date().toISOString().slice(0, 10)
+
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply lib/server/reservations-service.ts:427]: Fixed. The activation window is now anchored to `reservation.date` (the stored DB value) instead of a UTC-computed `today` string. `reservationStart` is built directly from `new Date(reservation.date + 'T' + reservation.start_time)`, removing the dependency on `new Date().toISOString().slice(0, 10)` for the date boundary check. A code comment documents the remaining UTC/local-time limitation as known tech debt for a future dedicated timezone migration.

File: lib/server/reservations-service.ts [hunk 2]
```diff
@@ -388,3 +394,91 @@ export async function updateReservationForSession(
...
+    .eq('table_id', tableId)
+    .eq('date', today)
+    .eq('user_id', userId)
+    .eq('status', 'pending')
+
+  if (side === 'inf') {
+    pendingQuery = pendingQuery.eq('surface', 'bottom')
+  } else {
+    pendingQuery = pendingQuery.eq('surface', 'top')
+  }
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply lib/server/reservations-service.ts:443]: Fixed. The pending and active lookups now use `.or('surface.eq.top,surface.is.null')` instead of `.eq('surface', 'top')`, so non-removable tables (stored with `surface = null`) are correctly matched alongside the top side of removable tables. Tests covering this path have been added and pass.

File: app/[locale]/check-in/[tableId]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    redirect(`/${locale}/login?returnUrl=/${locale}/check-in/${tableId}${sideParam === 'inf' ? '?side=inf' : ''}`)
+  }
+
+  const side = sideParam === 'inf' ? ('inf' as const) : undefined
+
+  let status: CheckInStatus = 'error'
+
+  try {
+    await activateReservationByTable(tableId, session.id, side)
+    status = 'activated'
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply app/[locale]/check-in/[tableId]/page.tsx]: Fixed. `page.tsx` is now a pure server component — it only handles the auth redirect and renders `<CheckInActivator>`. Activation has been moved to a new `'use client'` component (`components/check-in/check-in-activator.tsx`) that calls `POST /api/tables/[id]/activate` via `fetch` on mount, with an `AbortController` to prevent double-invocation in React 18 Strict Mode. The GET request no longer triggers any state-changing operation.

File: app/api/reservations/[id]/activate/route.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  const auth = await requireAuth(request)
+  if (auth instanceof NextResponse) return auth
+
+  try {
+    const { id: tableId } = await params
+    const { searchParams } = new URL(request.url)
+    const side = searchParams.get('side') === 'inf' ? 'inf' as const : undefined
+
+    const reservation = await activateReservationByTable(tableId, auth.session.id, side)
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply app/api/reservations/[id]/activate/route.ts:22]: Fixed. The activation endpoint has been moved to `app/api/tables/[id]/activate/route.ts`, consistent with the other table endpoints under `app/api/tables/[id]/`. The old `app/api/reservations/[id]/activate/` directory has been deleted entirely. The `CheckInActivator` client component calls the new route at `/api/tables/{tableId}/activate`.

File: components/check-in/check-in-result.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        return t('alreadyActive')
+      case 'too_early':
+        return t('tooEarly')
+      case 'too_late':
+        return t('tooLate')
+      case 'no_reservation':
+        return t('noReservation')
+      default:
+        return t('tooLate')
+    }
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply components/check-in/check-in-result.tsx:61]: Fixed. `getMessage()` now has an explicit `case 'error': return t('error')` branch, and the `default` case also returns `t('error')` instead of `t('tooLate')`. The `checkin.error` translation key has been added to both `messages/en.json` and `messages/es.json`.

File: __tests__/server/tables-service.test.ts [hunk 6]
```diff
@@ -88,3 +106,65 @@ describe('getTableAvailability', () => {
...
+    const result = await generateTableQrCode('table-abc')
+
+    expect(result).toMatch(/^data:image\/png;base64,/)
+  })
+
+  it('encodes the absolute URL with the tableId in the QR payload', async () => {
+    const { generateTableQrCode } = await loadTablesModules()
+
+    const result = await generateTableQrCode('table-abc')
+
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply __tests__/server/tables-service.test.ts:181]: Fixed. The QR generation logic has been migrated from `qrcode.toDataURL` to `qrcode.toBuffer` + Supabase Storage upload. The test now mocks `qrcode.toBuffer` as a spy and asserts it is called with the correct absolute URL (`${NEXT_PUBLIC_APP_URL}/check-in/${tableId}`), and separately asserts that `storage.from('table-qr-codes').upload()` is called with the expected path (`${tableId}.png`) and options. The returned value is asserted to be a Supabase Storage public URL, not a data URL.

File: __tests__/server/reservations-service.test.ts [hunk 7]
```diff
@@ -613,4 +667,156 @@ describe('reservations service', () => {
...
+      })
+      reservationsState.push(row)
+      return row
+    }
+
+    function makeStartTime(offsetMinutes: number): string {
+      const now = new Date()
+      now.setMinutes(now.getMinutes() - offsetMinutes)
+      const hh = String(now.getHours()).padStart(2, '0')
+      const mm = String(now.getMinutes()).padStart(2, '0')
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply __tests__/server/reservations-service.test.ts:708]: Fixed. The `activateReservationByTable` test suite now uses `vi.useFakeTimers()` with `vi.setSystemTime(new Date('2025-06-15T14:00:00Z'))` in `beforeEach` and `vi.useRealTimers()` in `afterEach`. All time-based constants (`FIXED_DATE`, `start_time`) are derived from this fixed timestamp, making the activation-window boundary assertions fully deterministic regardless of when the tests run.

File: components/check-in/check-in-activator.tsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+export function CheckInActivator({ tableId, side }: CheckInActivatorProps) {
+  const t = useTranslations('checkin')
+  const [status, setStatus] = useState<CheckInStatus | null>(null)
+
+  useEffect(() => {
+    const controller = new AbortController()
+    const url = `/api/tables/${encodeURIComponent(tableId)}/activate${side === 'inf' ? '?side=inf' : ''}`
+
+    fetch(url, { method: 'POST', signal: controller.signal })
+      .then(async (res) => {
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply components/check-in/check-in-activator.tsx]: Fixed. Replaced the raw `fetch()` call with `apiClient.post()` from `lib/api/client.ts`. The `apiClient` reads the `alea-csrf-token` cookie and injects the `x-csrf-token` header automatically on every non-GET request, so the double-submit CSRF check in `enforceMutationSecurity` will now pass correctly in production. The `AbortController` pattern was replaced with a `cancelled` flag since `apiClient` does not accept an abort signal — cleanup-on-unmount behavior is preserved.

File: app/api/cron/cancel-pending/route.ts [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import { NextResponse } from 'next/server'
+import { cancelExpiredPendingReservations } from '@/lib/server/reservations-service'
+
+export async function POST(request: Request) {
+  const auth = request.headers.get('Authorization')
+  if (!process.env.CRON_SECRET || auth !== `Bearer ${process.env.CRON_SECRET}`) {
+    return NextResponse.json({ error: 'Unauthorized' }, { status: 401 })
+  }
+  const cancelled = await cancelExpiredPendingReservations()
+  return NextResponse.json({ cancelled })
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply app/api/cron/cancel-pending/route.ts]: Fixed. Added a `GET` handler alongside the existing `POST`. Both delegate to a shared `handleCronRequest` function to avoid duplication. Vercel Cron will now hit the `GET` handler as expected; `POST` is kept as a secondary option for manual triggers.

File: supabase/migrations/20260411000000_fn_cancel_expired_pending_reservations.sql [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+-- start_time is stored as VARCHAR in 'HH:MM' format (e.g. '14:00').
+-- Casting 'HH:MM'::time gives a PostgreSQL TIME value which can be added to a DATE
+-- to produce a TIMESTAMP. Adding the 20-minute grace period and comparing with NOW()
+-- identifies reservations that should be marked as no_show.
+
+CREATE OR REPLACE FUNCTION public.cancel_expired_pending_reservations()
+RETURNS integer
+LANGUAGE plpgsql
+SECURITY DEFINER
+SET search_path = public, pg_temp
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply supabase/migrations/20260411000000_fn_cancel_expired_pending_reservations.sql]: Fixed. Changed `SET search_path = public, pg_temp` to `SET search_path = public, pg_catalog`, matching the existing pattern used by other `SECURITY DEFINER` helpers in the schema. `pg_temp` has been removed entirely to eliminate the privilege-escalation risk.

File: supabase/migrations/20260411000000_fn_cancel_expired_pending_reservations.sql [hunk 11]
```diff
@@ -0,0 +1,36 @@
+-- Migration: add function to cancel pending reservations not activated within 20 minutes
+--
+-- start_time is stored as VARCHAR in 'HH:MM' format (e.g. '14:00').
+-- Casting 'HH:MM'::time gives a PostgreSQL TIME value which can be added to a DATE
+-- to produce a TIMESTAMP. Adding the 20-minute grace period and comparing with NOW()
+-- identifies reservations that should be marked as no_show.
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply supabase/migrations/20260411000000_fn_cancel_expired_pending_reservations.sql]: Fixed. The comment has been updated to accurately reflect that `start_time` is a PostgreSQL `TIME NOT NULL` column. The misleading reference to `VARCHAR` and the cast explanation have been removed.

File: supabase/migrations/20260411000001_qr_storage_bucket.sql [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+-- Allow service role to upload/update/delete QR images
+CREATE POLICY IF NOT EXISTS "qr_codes_service_write"
+  ON storage.objects FOR INSERT
+  WITH CHECK (bucket_id = 'table-qr-codes' AND auth.role() = 'service_role');
+
+CREATE POLICY IF NOT EXISTS "qr_codes_service_update"
+  ON storage.objects FOR UPDATE
+  USING (bucket_id = 'table-qr-codes' AND auth.role() = 'service_role');
+
+CREATE POLICY IF NOT EXISTS "qr_codes_service_delete"
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply supabase/migrations/20260411000001_qr_storage_bucket.sql]: Fixed. Removed `IF NOT EXISTS` from all four `CREATE POLICY` statements — `IF NOT EXISTS` on policies is not valid in PostgreSQL 15 (it was added in PG 17). Idempotency is handled at the migration runner level.

File: supabase/migrations/20260411000001_qr_storage_bucket.sql [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+  USING (bucket_id = 'table-qr-codes');
+
+-- Allow service role to upload/update/delete QR images
+CREATE POLICY IF NOT EXISTS "qr_codes_service_write"
+  ON storage.objects FOR INSERT
+  WITH CHECK (bucket_id = 'table-qr-codes' AND auth.role() = 'service_role');
+
+CREATE POLICY IF NOT EXISTS "qr_codes_service_update"
+  ON storage.objects FOR UPDATE
+  USING (bucket_id = 'table-qr-codes' AND auth.role() = 'service_role');
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply supabase/migrations/20260411000001_qr_storage_bucket.sql]: Fixed. Added `WITH CHECK (bucket_id = 'table-qr-codes' AND auth.role() = 'service_role')` to the UPDATE policy so both the matched rows (`USING`) and the post-update state (`WITH CHECK`) are constrained, preventing a `service_role` UPDATE from moving an object to a different bucket.

File: lib/server/reservations-service.ts [hunk 14]
```diff
@@ -388,3 +394,112 @@ export async function updateReservationForSession(
...
+  const today = new Date().toISOString().slice(0, 10)
+
+  const admin = createSupabaseServerAdminClient()
+
+  let pendingQuery = (admin.from('reservations') as unknown as { select: (c: string) => ActivationAdminQuery })
+    .select(RESERVATION_COLUMNS)
+    .eq('table_id', tableId)
+    .eq('date', today)
+    .eq('user_id', userId)
+    .eq('status', 'pending')
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply lib/server/reservations-service.ts]: Acknowledged. The UTC/local-time limitation for `today` in `cancelExpiredPendingReservations` has been marked with a `// TODO(timezone): use venue local timezone instead of UTC` comment on the relevant line. A proper fix (venue-timezone config or a date-window query spanning yesterday/today) is tracked as known tech debt for a dedicated timezone migration. No runtime behavior changed in this PR.

File: supabase/migrations/20260411000001_qr_storage_bucket.sql [hunk 15]
```diff
@@ -0,0 +1,29 @@
+-- Create a public storage bucket for table QR code images
+INSERT INTO storage.buckets (id, name, public, file_size_limit, allowed_mime_types)
+VALUES (
+  'table-qr-codes',
+  'table-qr-codes',
+  true,
+  52428,  -- 50 KB max per file
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply supabase/migrations/20260411000001_qr_storage_bucket.sql]: Fixed. The comment on the `file_size_limit` line has been updated to read `52,428 bytes max per file` — matching the actual numeric value already in the code. The limit value itself is unchanged.

File: lib/server/reservations-service.ts [hunk 16]
```diff
@@ -388,3 +394,114 @@ export async function updateReservationForSession(
...
+): Promise<Reservation> {
+  // NOTE: We use today's local wall-clock date as a string anchor only for the
+  // initial DB query. The time-window check below uses the reservation's own
+  // stored date so the logic is self-consistent even if the request arrives
+  // near midnight. A full timezone fix should pass an IANA zone from the
+  // client or club config and use a proper date library (e.g. date-fns-tz).
+  // TODO(timezone): use venue local timezone instead of UTC — UTC may be off by
+  // 1-2 hours relative to the venue's wall-clock date near midnight.
+  const today = new Date().toISOString().slice(0, 10)
+
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply lib/server/reservations-service.ts]: Fixed. The inline comment now accurately says 'UTC date' instead of implying local wall-clock time. The code uses `new Date().toISOString().slice(0, 10)` which produces a UTC date string, and the comment now reflects that. No runtime behavior was changed.

File: app/api/tables/[id]/activate/route.ts [hunk 17]
```diff
@@ -0,0 +1,27 @@
+import { NextRequest, NextResponse } from 'next/server'
+import { requireAuth } from '@/lib/server/auth'
+import { toServiceErrorResponse } from '@/lib/server/http-error'
+import { activateReservationByTable } from '@/lib/server/reservations-service'
+import { enforceMutationSecurity, enforceRateLimit, RATE_LIMIT_POLICIES } from '@/lib/server/security'
+
+export async function POST(request: NextRequest, { params }: { params: Promise<{ id: string }> }) {
+  const securityError = enforceMutationSecurity(request)
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply app/api/tables/[id]/activate/route.ts:8]: Fixed. The PR description has been updated to reference `POST /api/tables/[id]/activate` (the actual implemented endpoint) instead of the old `POST /api/reservations/[id]/activate`. The test plan has also been updated to reflect the current state (231 passing tests, all ESLint warnings resolved).

File: components/check-in/check-in-activator.tsx [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+export function CheckInActivator({ tableId, side }: CheckInActivatorProps) {
+  const t = useTranslations('checkin')
+  const [status, setStatus] = useState<CheckInStatus | null>(null)
+
+  useEffect(() => {
+    let cancelled = false
+
+    const path = `/api/tables/${encodeURIComponent(tableId)}/activate${side === 'inf' ? '?side=inf' : ''}`
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply components/check-in/check-in-activator.tsx]: Fixed. The path was changed from `/api/tables/${tableId}/activate` to `/tables/${tableId}/activate`. Since `apiClient.post()` already prepends `NEXT_PUBLIC_API_URL` (default `/api`), the old path was resolving to `/api/api/tables/...` → 404 in production. The corrected path now follows the same convention used by all other hooks in `lib/hooks/use-admin.ts` (e.g. `/tables/${tableId}/qr`).

File: components/reservations/my-reservations-view.tsx [hunk 19]
```diff
@@ -15,6 +15,69 @@ import {
...
+              {reservation.roomName && reservation.tableName
+                ? `${reservation.roomName} · ${reservation.tableName}`
+                : reservation.tableName ?? reservation.tableId}
+            </span>
+            {reservation.surface && (
+              <span className="flex items-center gap-1 text-xs text-muted-foreground">
+                <Layers className="h-3 w-3" aria-hidden="true" />
+                {reservation.surface === 'top' ? 'Superior' : 'Inferior'}
+              </span>
+            )}
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply components/reservations/my-reservations-view.tsx:50]: Fixed. Replaced the hardcoded Spanish strings `'Superior'`/`'Inferior'` with `tt('surfaceTop')`/`tt('surfaceBottom')` using a `useTranslations('tables')` hook inside `ReservationCard`. Both `en.json` (`Top (lid on)`/`Bottom (gaming board)`) and `es.json` (`Superior (tapa puesta)`/`Inferior (tablero de juego)`) already had these keys, so no i18n additions were needed.

File: lib/server/tables-service.ts [hunk 20]
```diff
@@ -20,6 +21,84 @@ function toGameTable(row: TableRow): GameTable {
...
+export async function generateTableQrCode(tableId: string): Promise<string> {
+  if (!/^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/i.test(tableId)) {
+    serviceError('Invalid table ID', 400)
+  }
+  const appUrl = (process.env.NEXT_PUBLIC_APP_URL ?? '').replace(/\/$/, '')
+  if (!appUrl) serviceError('NEXT_PUBLIC_APP_URL is not set — cannot generate QR code URL', 500)
+  const url = `${appUrl}/check-in/${tableId}`
+  const admin = createSupabaseServerAdminClient()
+  return uploadQrCodeToStorage(admin, url, `${tableId}.png`)
+}
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply lib/server/tables-service.ts:55]: Fixed. `createTableEntry` in `lib/server/rooms-service.ts` now calls `regenerateQrCodes(tableId)` immediately after the table row is inserted. The returned `qr_code` (and `qr_code_inf` for `removable_top` tables) is stored on the table record before `toGameTable()` returns, so every new table gets its QR code automatically. If `NEXT_PUBLIC_APP_URL` is unset or the Storage upload fails, the error is caught and logged — table creation succeeds and the admin can regenerate the QR manually.

File: supabase/migrations/20260411000001_qr_storage_bucket.sql (hunk 15 above)
[the remark this answers is not in the document]
Comment (Oiranca) [reply supabase/migrations/20260411000001_qr_storage_bucket.sql]: Fixed. Updated the inline comment to `52,428 bytes max per file`, matching the exact configured value of the `file_size_limit` field. The comment now accurately documents what is actually set.

File: lib/server/reservations-service.ts (hunk 16 above)
[the remark this answers is not in the document]
Comment (Oiranca) [reply lib/server/reservations-service.ts]: Fixed. Updated the inline comment to accurately reflect that the query is anchored to today's UTC date. The misleading reference to local wall-clock date has been removed.

File: lib/server/tables-service.ts [hunk 21]
```diff
@@ -20,6 +21,84 @@ function toGameTable(row: TableRow): GameTable {
   }
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply lib/server/tables-service.ts:22]: Fixed. Added `qrCodeInf: row.qr_code_inf ?? null` to `toGameTable()` in `tables-service.ts`, matching the pattern already used in `rooms-service.ts`. All code paths that select `qr_code_inf` now consistently map it through to `GameTable`.

File: lib/server/reservations-service.ts [hunk 22]
```diff
@@ -388,3 +394,114 @@ export async function updateReservationForSession(
...
+type AdminRpcClient = {
+  rpc: (fn: string) => Promise<{ data: unknown; error: { message: string } | null }>
+}
+
+export async function cancelExpiredPendingReservations(): Promise<number> {
+  const admin = createSupabaseServerAdminClient() as unknown as AdminRpcClient
+  const { data, error } = await admin.rpc('cancel_expired_pending_reservations')
+  if (error) serviceError('Internal server error', 500)
+  return (data as number) ?? 0
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply lib/server/reservations-service.ts]: Fixed. Added `cancel_expired_pending_reservations: { Args: Record<PropertyKey, never>; Returns: number }` to the `Functions` block for the `public` schema in `lib/supabase/types.ts`, alongside the existing `is_admin` entry. The local `AdminRpcClient` workaround type has been removed entirely. `admin.rpc('cancel_expired_pending_reservations')` is now fully typed via the generated Database types — no cast needed.

File: app/api/cron/cancel-pending/route.ts [hunk 23]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+// Vercel Cron sends GET requests to the configured path.
+export async function GET(request: Request) {
+  return handleCronRequest(request)
+}
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply app/api/cron/cancel-pending/route.ts:16]: Fixed. Added a `POST` export delegating to the same `handleCronRequest` function. The route now supports both GET (for Vercel Cron) and POST (for manual triggers). Tests cover both methods: 4 GET tests + 4 POST tests with identical auth-validation patterns.

File: supabase/migrations/20260410000001_reservation_status_columns.sql [hunk 24]
```diff
@@ -0,0 +1,11 @@
+-- Migration: set pending as default reservation status, add activated_at and qr_code_inf
+-- Must run in a separate transaction from the ADD VALUE statements above.
+
+-- Change default status for new reservations from 'active' to 'pending'
+ALTER TABLE public.reservations ALTER COLUMN status SET DEFAULT 'pending';
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply supabase/migrations/20260410000001_reservation_status_columns.sql:5]: Fixed. The `pending` default is intentional — all new reservations start as pending and become active only after QR check-in. The availability queries in `getTableAvailability()` and `getRoomTablesAvailability()` now filter `.in('status', ['active', 'pending'])`, treating pending reservations as blocking exactly like conflict-detection already does. Users will no longer see booked-but-unactivated slots as available.

File: lib/server/tables-service.ts [hunk 25]
```diff
@@ -20,6 +21,84 @@ function toGameTable(row: TableRow): GameTable {
...
+  if (updateError) {
+    serviceError('Internal server error', 500)
+  }
+
+  return { qr_code, qr_code_inf }
+}
+
 export async function getTableAvailability(tableId: string, date?: string | null) {
   const supabase = await createSupabaseServerClient()
   const tableResult = await supabase
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply lib/server/tables-service.ts:105]: Fixed. Updated `getTableAvailability()` in `tables-service.ts` to filter `.in('status', ['active', 'pending'])` instead of `.eq('status', 'active')`. Applied the same fix to `getRoomTablesAvailability()` in `rooms-service.ts` and updated the `ReservationsByTableClient` local type to match the new query chain. A test confirming pending reservations occupy time slots has been added to `__tests__/server/tables-service.test.ts`.
